app/models/repository.rb

```diff
@@ -87,14 +87,24 @@ def scm
     @scm
   end
 
-  def vendor
-    self.class.vendor
+  def self.scm_config
+    scm_adapter_class.config
   end
 
-  def supported_types
+  def self.available_types
     []
```

Comment:
I found it quite confusing that `available_types` and `disabled_types` are unrelated to eachother.

As far as I can see you are building the link in the subclass.

How about this idea:

````ruby
def self.available_types
  supported_types - disabled_types
end
````
With `supported_types` being defined by the subclass and representing the theoretical set of possible types.

Tell me what you think ^^